server/src/main/java/org/elasticsearch/persistent/PersistentTasksService.java

```diff
@@ -192,7 +214,13 @@ public void onClusterServiceClose() {
             public void onTimeout(TimeValue timeout) {
                 listener.onTimeout(timeout);
             }
-        }, clusterState -> predicate.test(PersistentTasksCustomMetadata.getTaskWithId(clusterState, taskId)), timeout, logger);
+        },
+            clusterState -> predicate.test(
+                PersistentTasksCustomMetadata.getTaskWithId(clusterState.metadata().getProject(projectId), taskId)
+            ),
```

Comment:
I think we should handle the case where the project is deleted and use `null` in place, assuming predicate can handle `null` (I think it should or it is an issue already).